### PR TITLE
Simplify function handleRequest

### PR DESCRIPTION
**Pull request by trivikr**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by trivikr** [comment]
I don't see any significant impact on benchmarks

Benchmark with PR:
```
> fastify@0.43.0 benchmark /Users/trivikr/workspace/fastify
> npx concurrently -k -s first "node ./examples/simple.js" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"

[1] Running 5s test @ http://localhost:3000/
[1] 100 connections with 10 pipelining factor
[1]
[0] server listening on 3000
[1] Stat         Avg     Stdev   Max
[1] Latency (ms) 5.75    19.07   510.05
[1] Req/Sec      16768.8 4117.93 19231
[1] Bytes/Sec    2.48 MB 605 kB  2.87 MB
[1]
[1] 84k requests in 5s, 12.5 MB read
[1] 143 errors (0 timeouts)
[1] npx autocannon -c 100 -d 5 -p 10 localhost:3000/ exited with code 0
--> Sending SIGTERM to other processes..
[0] node ./examples/simple.js exited with code null
```

Benchmark on master:
```
> fastify@0.43.0 benchmark /Users/trivikr/workspace/fastify
> npx concurrently -k -s first "node ./examples/simple.js" "npx autocannon -c 100 -d 5 -p 10 localhost:3000/"

[1] Running 5s test @ http://localhost:3000/
[1] 100 connections with 10 pipelining factor
[1]
[0] server listening on 3000
[1] Stat         Avg     Stdev   Max
[1] Latency (ms) 5.73    18.74   346.97
[1] Req/Sec      17004   2889.81 18933
[1] Bytes/Sec    2.54 MB 436 kB  2.82 MB
[1]
[1] 85k requests in 5s, 12.7 MB read
[1] 94 errors (0 timeouts)
[1] npx autocannon -c 100 -d 5 -p 10 localhost:3000/ exited with code 0
--> Sending SIGTERM to other processes..
[0] node ./examples/simple.js exited with code null
```

**Comment by mcollina** [comment]
I'm kind of -1 with this change, because it increase the number of inlined functions, reducing what is available to the application (functions can be inlined up to 5 level deep).

**Comment by trivikr** [comment]
Removed inline functions as recommended by @mcollina and @delvedor

File: lib/handleRequest.js [hunk 1]
```diff
@@ -14,14 +14,11 @@ function handleRequest (req, res, params, context) {
...
   var reply = new context.Reply(res, context, request)
-
-  if (method === 'GET' || method === 'HEAD') {
-    return handler(reply)
-  }
-
   var contentType = headers['content-type']
 
-  if (method === 'POST' || method === 'PUT' || method === 'PATCH') {
+  if (method === 'GET' || method === 'HEAD') {
```
Comment (nwoltman) [inline lib/handleRequest.js]: I would keep this first `if` statement where it is, otherwise the `content-type` header is read when it doesn't need to be.

**Comment by trivikr** [comment]
Discarding this PR, as this code change will end up replacing `return` statements with `if-else` after incorporating suggestions from review comments.

**Pull request closed without merging**